docs/collecting-data/collecting-from-own-applications/python-tracker/subject/index.md

```diff
@@ -0,0 +1,236 @@
+---
+title: "Subject"
+date: "2023-06-14"
+sidebar_position: 40
+---
+
+You may have additional information about the user (i.e. subject) performing the action or the environment in which the user has performed the action. Some of that additional data can be sent into Snowplow with each event as part of the subject class.
+
+You can create a subject like this:
+
+```python
+from snowplow_tracker import Subject
+s = Subject()
+```
+
+The Subject class has a set of `set_...()` methods to attach extra data relating to the user:
+
+- [`set_platform`](#change-the-trackers-platform-withset_platform)
+- [`set_user_id`](#set-user-id-withset_user_id)
+- [`set_screen_resolution`](#set-screen-resolution-with-set_screen_resolution)
+- [`set_viewport`](#set-viewport-dimensions-withset_viewport)
+- [`set_color_depth`](#set-color-depth-withset_color_depth)
+- [`set_timezone`](#set-timezone-withset_timezone)
+- [`set_lang`](#set-the-language-withset_lang)
+
+If you initialize a `Tracker` instance without a subject, a default `Subject` instance will be attached to the tracker. You can access that subject like this:
+
+```python
+t = Tracker(my_emitter)
+t.subject.set_platform("mob").set_user_id("user-12345").set_lang("en")
+```
+
+We will discuss each of these in turn below:
+
+### Change the tracker's platform with `set_platform`
+
+The default platform is "pc". You can change the platform the subject is using by calling for example:
+
+```python
+s.set_platform('mob')
+```
+
+For a full list of supported platforms, please see the [Snowplow Tracker Protocol](/docs/collecting-data/collecting-from-own-applications/snowplow-tracker-protocol/index.md#application-parameters).
+
+### Set user ID with `set_user_id`
+
+You can set the user ID to any string:
+
+```python
+s.set_user_id("user_id")
+```
+
+### Set screen resolution with `set_screen_resolution`
+
+If your Python code has access to the device's screen resolution, then you can pass this in to Snowplow too. 
+Both numbers should be positive integers; note the order is width followed by height. For example:
+
+```python
+s.set_screen_resolution(1366, 768)
+```
+
+### Set viewport dimensions with `set_viewport`
+
+If your Python code has access to the viewport dimensions, then you can pass this in to the subject too. Both numbers should be positive integers; note the order is width followed by height. For example:
+
+```python
```

Comment:
These are basically the same paragraph but in one you say it can be passed to the subject, the other says to Snowplow, can we be consistent throughout the whole page please

Reply:
To be honest most of these say the same thing of "If you have access to X, you can set it with Y, it should be of type Z". Is there a better way to rework this to reduce duplication?